server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -352,6 +352,19 @@ public void testFilteredBooleanSetting() {
         assertNull(e.getCause());
     }
 
+    private enum TestEnumSetting {
+        ON,
+        OFF
+    }
+
+    public void testThrowsIllegalArgumentExceptionOnInvalidEnumSetting() {
+        Setting setting = Setting.enumSetting(TestEnumSetting.class, "foo", TestEnumSetting.ON, Property.Filtered);
+        final Settings settings = Settings.builder().put("foo", "bar").build();
+
+        final IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> setting.get(settings));
+        assertNull(e.getCause());
```

Comment:
This seems like an odd test, to check that the exception has no cause. Can we instead check part of the message to be sure it in the IAE we expect?